packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -296,11 +325,70 @@ export default createRule<Options, MessageId>({
   },
 });
 
+function isPromiseArray(checker: ts.TypeChecker, node: ts.Node): boolean {
+  const type = checker.getTypeAtLocation(node);
+  if (checker.isArrayType(type)) {
+    const arrayType = checker.getTypeArguments(type)[0];
+    if (isPromiseLikeType(checker, arrayType)) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+function isPromiseLikeType(checker: ts.TypeChecker, type: ts.Type): boolean {
```

Comment:
1. I generally struggled to find out info about the TS Compiler API. If anyone knows where to find details please link me to it.

2. I made a new utility function here in order to check whether the `T` in an `Array<T>` is Promise-like, basing it off the existing function (now renamed to isPromiseLikeNode) . I didn't use that function because it depends on the specific location of the node it's inspecting, not just the `ts.Type` and so I couldn't figure out if it was possible to use it for checking an `Array<T>` (See item 1). Please advise if this is a bad idea. I'm guessing it won't have 100% test coverage as written, since I didn't bother yet writing test cases for all the NonFunctionHandlerThenable cases. If it turns out this is the right way to go, I can go ahead and fill those out.